code/obj/flock/furniture.dm

```diff
@@ -298,3 +310,9 @@
 		<br><span class='bold'>###=-</span></span>"}
 	else
 		return null // give the standard description
+
+/obj/grille/flock/attackby(obj/item/W as obj, mob/user as mob)
+	if (istype(user, /mob/living/critter/flock/drone))
+		boutput(user, "<span class='alert'>The grip tool refuses to allow damage to a barricade, jamming briefly.</span>")
+		return
+	..()
```

Comment:
While it might be overkill, you could componentize this check and just apply them to every flock converted subtype.

Reply:
Sorry, not really familiar with that, if you explain I could look into it